src/spock_group.c

```diff
@@ -187,7 +187,12 @@ spock_group_attach(Oid dbid, Oid node_id, Oid remote_node_id)
 	SpockGroupEntry *e;
 	bool		found;
 
+	if (!SpockGroupHash)
+		elog(ERROR, "spock_group_attach: SpockGroupHash not initialized");
+
 	e = (SpockGroupEntry *) hash_search(SpockGroupHash, &key, HASH_ENTER, &found);
+	if (!e)
+		elog(ERROR, "spock_group_attach: hash_search failed");
 	if (!found)
 	{
```

Comment:
there is no need for this 2nd check. !found is covering it when a new entry is added to the hash. also when existing entry is found, that is returned. hash_search errors out if there is any issues, it would not return NULL for HASH_ENTER.